lib/AFS/CellCC/Dump.pm

```diff
@@ -185,77 +185,82 @@ _calc_incremental($$) {
-                   description => "Remote volume appears to be up to date; skipping sync");
-        return;
-    }
-
-    DEBUG "got lastupdate time $lastupdate for volname $volname";
+    update_jobs(\@jobs,
+                from_state => $prev_state,
+                to_state => $state,
+                timeout => 300,
+                description => "Checking local volume state");
 
     my $partition;
+    # every single entry from jobs has the same src cell
     ($server, $partition) = find_volume(name => $volname,
                                         type => 'RO',
                                         server => $server,
-                                        cell => $job->{src_cell});
+                                        cell => $jobs[0]->{src_cell});
 
-    my $dump_size = _get_size($job, $volname, $server, $partition, $lastupdate);
+    # the first argument of _get_size is not used
+    my $dump_size = _get_size($jobs[0], $volname, $server, $partition, $lastupdate);
     DEBUG "got dump size $dump_size for volname $volname";
 
     if (!scratch_ok($prev_state, $dump_size,
                     config_get('dump/scratch-dir'),
-                    config_get('dump/scratch-minfree'), $job)) {
+                    config_get('dump/scratch-minfree'), @jobs)) {
         return;
     }
 
-    update_job(jobid => $job->{jobid},
-               dvref => \$job->{dv},
-               from_state => $state,
-               vol_lastupdate => $lastupdate,
-               timeout => 120,
-               description => "Starting to dump volume");
-    $job->{vol_lastupdate} = $lastupdate;
+    db_rw(sub($) {
+        my ($sub_dbh) = @_;
+        my $descr;
+        for my $i (0 .. $#jobs) {
+            if ($i == 0) {
+                $descr = "Starting to dump volume";
+            } else {
+                $descr = "Waiting for dump from job $jobs[0]->{jobid}";
+            }
+            update_job(dbh => $sub_dbh,
+                       jobid => $jobs[$i]->{jobid},
+                       dvref => \$jobs[$i]->{dv},
+                       from_state => $state,
+                       vol_lastupdate => $lastupdate,
+                       timeout => 120,
+                       description => $descr);
+        }
+    });
 
-    my $stderr_fh = File::Temp->new(TEMPLATE => "cccdump_job$job->{jobid}_XXXXXX",
+    my $ids = join( '_', get_jobs_info("jobid", @jobs) );
+    my $stderr_fh = File::Temp->new(TEMPLATE => "cccdump_jobs{$ids}_XXXXXX",
                                     TMPDIR => 1, SUFFIX => '.stderr');
 
     # Determine a filename where we can put our dump blob
-    my $dump_fh = File::Temp->new(DIR => config_get('dump/scratch-dir'),
-                                  TEMPLATE => "cccdump_job$job->{jobid}_XXXXXX",
-                                  SUFFIX => '.dump');
+    my @dump_fhs;
+    my $pjob = $jobs[0]->{jobid};
+    for my $job (@jobs) {
+        my $dump_fh = File::Temp->new(DIR => config_get('dump/scratch-dir'),
```

Comment:
I don't think there's a reason to create these tempfiles so early. it's not a big deal, but it means these temp files stay around for longer while they're not needed (which means a higher chance of random crud being left around, since these files aren't recorded in the database yet). we could leave this as normal, and create the temp files for the links at the same time as creating the links.